api/src/main/java/edu/wpi/first/shuffleboard/api/data/DataType.java

```diff
@@ -9,52 +11,58 @@
  *
  * @see DataTypes
  */
-public interface DataType<T> {
+public abstract class DataType<T> {
+
+  private final String name;
+  private final Class<T> javaClass;
+
+  protected DataType(String name, Class<T> javaClass) {
+    this.name = name;
+    this.javaClass = javaClass;
+  }
 
   /**
    * Gets the name of this data type.
    */
-  String getName();
+  public final String getName() {
+    return name;
+  }
+
+  public final Class<T> getJavaClass() {
+    return javaClass;
+  }
 
   /**
    * Gets the default value of this data type, eg 0 for numbers or an empty String for text.
    */
-  T getDefaultValue();
+  public abstract T getDefaultValue();
 
   /**
    * Checks if this data type is complex or not. This is class-intrinsic.
    */
-  boolean isComplex();
+  public abstract boolean isComplex();
 
-  /**
-   * Gets the data type most closely associated with the given Java type.
-   */
-  static DataType<?> forJavaType(Class<?> type) {
-    if (type == String.class) {
-      return DataTypes.String;
-    } else if (Number.class.isAssignableFrom(type)
-        || type == double.class || type == int.class || type == long.class) {
-      return DataTypes.Number;
-    } else if (type == Boolean.class || type == boolean.class) {
-      return DataTypes.Boolean;
-    } else if (type == String[].class) {
-      return DataTypes.StringArray;
-    } else if (type == double[].class || type == Double[].class) {
-      return DataTypes.NumberArray;
-    } else if (type == boolean[].class || type == Boolean[].class) {
-      return DataTypes.BooleanArray;
-    } else if (type == byte[].class || type == Byte[].class) {
-      return DataTypes.RawBytes;
-    } else {
-      return DataTypes.Unknown;
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == null) {
+      return false;
     }
```

Comment:
Usually, you have a check for this immediately after the null check as an optimization:
```java
if (this == obj) return true;
```